tests/unit/lib/utils/test_osutils.py

```diff
@@ -128,3 +128,12 @@ def test_must_copy_if_symlink_fails(self, patched_copy_tree, pathced_os, patched
 
         pathced_os.symlink.assert_called_once()
         patched_copy_tree.assert_called_with(source_path, destination_path)
+
+
+class Test_rmtree_callback(TestCase):
+    @patch("samcli.lib.utils.osutils.os")
+    def test_rmtree_callback_raises_file_not_found_exception(self, patched_os):
+        patched_os.chmod.side_effect = FileNotFoundError("File not found")
+
+        with self.assertRaises(FileNotFoundError):
+            rmtree_callback(Mock(), "/some/path", Mock())
```

Comment:
nit: I know we are mocking here, does the `/some/path` being UNIX path matter at all?